messaging/src/main/java/org/axonframework/eventhandling/deadletter/jpa/JpaSequencedDeadLetterQueue.java

```diff
@@ -0,0 +1,716 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.lang.invoke.MethodHandles;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import javax.annotation.Nonnull;
+import javax.persistence.EntityManager;
+import javax.persistence.NoResultException;
+
+import static org.axonframework.common.BuilderUtils.assertNonNull;
+import static org.axonframework.common.BuilderUtils.assertThat;
+
+/**
+ * JPA-backed implementation of the {@link SequencedDeadLetterQueue}, used for storing dead letters containing
+ * {@link EventMessage Eventmessages} as a {@link DeadLetterEntry}.
+ * <p>
+ * Keeps the insertion order intact by saving an incremented index within each sequence, backed by the
+ * {@link DeadLetterEntry#getIndex()} property.
+ * <p>
+ * When processing an item, single execution across all applications is guaranteerd by setting the
+ * {@link DeadLetterEntry#getProcessingStarted()} property, locking other processes out of the sequence for the
+ * configured {@code claimDuration} (30 seconds by default).
+ * <p>
+ * Configuring a chain of upcasters is not supported.
+ *
+ * @param <M> An implementation of {@link Message} contained in the {@link DeadLetter dead-letters} within this queue.
+ * @author Mitchell Herrijgers
+ * @since 4.6.0
+ */
+public class JpaSequencedDeadLetterQueue<M extends EventMessage<?>> implements SequencedDeadLetterQueue<M> {
+
+    private static final Logger logger = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+
+    private final String processingGroup;
+    private final EntityManagerProvider entityManagerProvider;
+    private final List<DeadLetterJpaConverter<EventMessage<?>>> converters;
+    private final TransactionManager transactionManager;
+    private final int maxSequences;
+    private final int maxSequenceSize;
+    private final Serializer serializer;
+    private final Duration claimDuration;
+
+    protected <T extends EventMessage<?>> JpaSequencedDeadLetterQueue(Builder<T> builder) {
+        builder.validate();
+        this.processingGroup = builder.processingGroup;
+        this.maxSequences = builder.maxSequences;
+        this.maxSequenceSize = builder.maxSequenceSize;
+        this.entityManagerProvider = builder.entityManagerProvider;
+        this.transactionManager = builder.transactionManager;
+        this.serializer = builder.serializer;
+        this.converters = builder.converters;
+        this.claimDuration = builder.claimDuration;
+    }
+
+    /**
+     * Creates a new builder, capable of building a {@link JpaSequencedDeadLetterQueue} according to the provided
+     * configuration. Note that the {@link Builder#processingGroup(String)} is mandatory for the queue to work
+     * correctly.
```

Comment:
I would expect the JavaDoc to state which fields the user is inclined to configure.